### PR TITLE
Suppress color output on aws

### PR DESCRIPTION
**Pull request by hannes-steffenhagen-diffblue**
Color codes don't show up correctly on AWS CodeBuild console, so we're disabling them to make the output more legible (they're still on by default!). Having the ability to turn color codes off can be useful in other situations (like scripts) as well.

- [ ] Each commit message has a non-empty body, explaining why the change was made.
- [ ] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [ ] The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- [ ] My commit message includes data points confirming performance improvements (if claimed).
- [x] My PR is restricted to a single feature or bugfix.
- [x] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

File: buildspec-linux-clang.yml [hunk 1]
```diff
@@ -1,5 +1,10 @@
 version: 0.2
 
+env:
+  variables:
+    # CodeBuild console doesn't display color codes correctly
+    TESTPL_SUPPRESS_COLOR_OUTPUT: 1
```
Comment (smowton) [inline buildspec-linux-clang.yml]: I'd recommend `TESTPL_COLOR_OUTPUT` rather than `SUPPRESS_` with an implicit negation, otherwise lgtm

**Review by smowton — APPROVED** [review]

File: buildspec-linux-clang.yml (hunk 1 above)
Comment (chrisr-diffblue) [reply buildspec-linux-clang.yml]: +1

File: regression/test.pl [hunk 2]
```diff
@@ -289,6 +301,8 @@ ($$$$)
              as runs with different suffixes will operate independently and keep
              independent logs.
   -f         forward the test name to CMD
+  -S <0|1>   (un-)suppress color output, default 0 if TESTPL_SUPPRESS_COLOR_OUTPUT
```
Comment (chrisr-diffblue) [inline regression/test.pl]: Rather than -S 0/1 to turn this on/off, could you have say "-P" for "plain output" - and just default to colour unless "-P" or TESTPL_COLOR_OUTPUT=0 ?

Comment (hannes-steffenhagen-diffblue) [reply regression/test.pl]: @chrisr-diffblue I wanted a way to force it back on through command line arguments if `TESTPL_SUPPRESS_COLOR_OUTPUT=1`

**Comment by hannes-steffenhagen-diffblue** [comment]
@chrisr-diffblue @smowton I changed the option name to `--[no]color` and the env var name to `TESTPL_COLOR_OUTPUT`, and flipped the logic to match the new names

File: regression/test.pl [hunk 3]
```diff
@@ -290,6 +300,11 @@ ($$$$)
              independent logs.
   -f         forward the test name to CMD
 
+  --[no]color enable/disable color output; enabled by default unless
+              TESTPL_COLOR_OUTPUT is set, in which case it is enabled by
+              default if its value is nonempty and nonzero, and enabled
+              by default otherwise.
+
```
Comment (chrisr-diffblue) [inline regression/test.pl:306]: I may just be being particularly slow today, but I find that message a little difficult to parse :-) Maybe a clearer description might be something like:

enabled by default, unless the TESTPL_COLOR_OUTPUT environment variable is set to a nonempty, nonzero value. This option overrides any value of TESTPL_COLOR_OUTPUT

what do you think?

File: regression/test.pl [hunk 4]
```diff
@@ -324,9 +339,13 @@ ($$$$)
...
 $Getopt::Std::STANDARD_HELP_VERSION = 1;
-our ($opt_c, $opt_e, $opt_f, $opt_i, $opt_j, $opt_n, $opt_p, $opt_h, $opt_C, $opt_T, $opt_F, $opt_K, $opt_s, %defines, @include_tags, @exclude_tags); # the variables for getopt
-GetOptions("D=s" => \%defines, "X=s" => \@exclude_tags, "I=s" => \@include_tags);
-getopts('c:efi:j:nphCTFKs:') or &main::HELP_MESSAGE(\*STDOUT, "", $main::VERSION, "");
+our ($opt_c, $opt_e, $opt_f, $opt_i, $opt_j, $opt_n, $opt_p, $opt_h, $opt_C, $opt_T, $opt_F, $opt_K, $opt_s, $opt_S, %defines, @include_tags, @exclude_tags); # the variables for getopt
+$color_output_enabled = 1;
+if (exists $ENV{'TESTPL_COLOR_OUTPUT'}) {
+  $color_output_enabled = $ENV{'TESTPL_COLOR_OUTPUT'};
+}
+GetOptions("D=s" => \%defines, "X=s" => \@exclude_tags, "I=s" => \@include_tags, 'color!' => \$color_output_enabled);
```
Comment (chrisr-diffblue) [inline regression/test.pl:350]: Is it worth adding a comment to explain that the test/setting of value from the ENV *must* come before the call to GetOptions to preserve the behaviour that the command line option overrides the ENV variable?

File: regression/test.pl (hunk 3 above)
Comment (hannes-steffenhagen-diffblue) [reply regression/test.pl:306]: I think it might be hard to parse because I mistyped the latter half of the sentence, it was meant to be

> and disabled by default otherwise.

instead of

> and enabled by default otherwise.

File: regression/test.pl (hunk 4 above)
Comment (hannes-steffenhagen-diffblue) [reply regression/test.pl:350]: done

File: regression/test.pl (hunk 3 above)
Comment (hannes-steffenhagen-diffblue) [reply regression/test.pl:306]: I've simplified the text, should be easier to parse now

**Review by chrisr-diffblue — APPROVED** [review]
